packages/template-retail-react-app/app/contexts/index.js

```diff
@@ -27,23 +29,103 @@ import PropTypes from 'prop-types'
  *
  * import {useCategories} from './hooks'
  *
- * const {categories, setCategories} = useCategories()
+ * const {root, itemsKey, fetchCategoryNode} = useCategories()
  *
  */
 export const CategoriesContext = React.createContext()
-export const CategoriesProvider = ({categories: initialCategories = {}, children}) => {
-    const [categories, setCategories] = useState(initialCategories)
+export const CategoriesProvider = ({treeRoot = {}, children, locale}) => {
+    const itemsKey = 'categories'
+    const DEFAULT_ROOT_CATEGORY = 'root'
+    const LOCAL_STORAGE_PREFIX = `pwa-kit-cat-`
+
+    const api = useCommerceAPI()
+    const [root, setRoot] = useState({
+        // map over the server-provided cat
+        ...treeRoot[DEFAULT_ROOT_CATEGORY],
+        [itemsKey]: treeRoot[DEFAULT_ROOT_CATEGORY]?.[itemsKey]?.map((item) => ({
+            ...item,
+            loaded: false
+        }))
+    })
+
+    const fetchCategoryNode = async (id, levels = 1) => {
+        // return early if there's one that is less than stale time
+        const storageString = window?.localStorage?.getItem(
+            `${LOCAL_STORAGE_PREFIX}${id}-${locale}`
+        )
+        if (Date.now() < storageString?.fetchTime + STALE_TIME) {
+            return storageString
+        }
+        let res
+        try {
+            res = await api.shopperProducts.getCategory({
+                parameters: {
+                    id,
+                    levels
+                }
+            })
+        } catch (e) {
+            console.error(e)
+        }
+        return res
```

Comment:
```suggestion
        return await api.shopperProducts.getCategory({
                parameters: {
                    id,
                    levels
                }
            })
```
Why is the error logged and ignored here? Seems like it should be allowed to bubble up to whatever error handling layer we have.